### DISCUSSION TITLE
Is it possible to query unique AudioFormats?

### DISCUSSION THREAD

**Discussion by m7a** [MusicPlayerDaemon/MPD#2099] [Q&A] [answered]
Hello MPD community,

I am working on my own client and want to create a *search* function for the users. One of the features I have envisioned it to have is a search by sample rate and bit depth. This is going to be mostly useful for testing purposes (debug audio setup issues by playing some music from known sample rate and frequency, checking the influence of resampling etc).

In order for the user to know that they can search for there, I would like to present the range of possible values in the UI. It may look something like this:

~~~
┌────────────[ MY SEARCH SCREEN ]────────────────┐
│                                                │
│ Artist             __________________________  │
│                                                │
│ Album              __________________________  │
│                                                │
│ Title              __________________________  │
│                                                │
│ Year               ___________ to ___________  │
│                    min. 1986      max. 2017    │
│                                                │
│ Sample rate [Hz]   ___________ to ___________  │
│                    min. 44100     max. 192000  │
│                                                │
│ Bit depth          ___________ to ___________  │
│                    min. 16        max. 24      │
│                                                │
│                    [SEARCH]  [RESET]           │
└────────────────────────────────────────────────┘
~~~

In this view, the `_____` would correspond to input fields and the sensible ranges of the values are ideally populated from values which are queried from the database.

Is there any way to query this efficiently?

My current best bet is to come up with a list of common formats and query by means of the `AudioFormat` filter whether they are present in the database or not, but this seems it could easily become out of date with what is actually used and also becomes less efficient as the list of known formats grows...?

Thanks in advance
Linux-Fan (@m7a)

**Comment by MaxKellermann** [comment, marked as answer]
Not possible currently.

**Reply by m7a** [reply]
I see, thakns for the quick confirmation :smile:
